src/runtime/schema.ts

```diff
@@ -217,13 +219,16 @@ export class Schema {
   toInlineSchemaString(options?: {hideFields?: boolean}): string {
     const names = this.names.join(' ') || '*';
     const fields = Object.entries(this.fields).map(Schema.fieldToString).join(', ');
-    return `${names} {${fields.length > 0 && options && options.hideFields ? '...' : fields}}`;
+    return `${names} {${fields.length > 0 && options && options.hideFields ? '...' : fields}}${this.refinement ? this.refinement.toString() : ''}`;
```

Comment:
I've moved the to String for the refinement to the schema level to ensure that it is always available (and reduce the particle handle connection's complexity.